File: pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/endpoints/message_actions/RemoveMessageAction.java
package com.pubnub.api.endpoints.message_actions;

import com.pubnub.api.endpoints.Endpoint;
import com.pubnub.api.models.consumer.message_actions.PNRemoveMessageActionResult;

public interface RemoveMessageAction extends Endpoint<PNRemoveMessageActionResult> {
    RemoveMessageAction channel(String channel);

    RemoveMessageAction messageTimetoken(Long messageTimetoken);

    RemoveMessageAction actionTimetoken(Long actionTimetoken);
}

File: pubnub-kotlin/src/main/kotlin/com/pubnub/api/endpoints/message_actions/RemoveMessageAction.kt
package com.pubnub.api.endpoints.message_actions

import com.pubnub.api.Endpoint
import com.pubnub.api.models.consumer.message_actions.PNRemoveMessageActionResult

/**
 * @see [PubNub.removeMessageAction]
 */
interface RemoveMessageAction :
    Endpoint<PNRemoveMessageActionResult> {
    val channel: String
    val messageTimetoken: Long
    val actionTimetoken: Long
}
